app/services/entitlement_read_service.py

```
@@ -0,0 +1,557 @@
+    if product_id is not None:
+        terminal = terminal.filter(EntitlementEvent.product_id == product_id)
+
+    terminal_count = terminal.scalar() or 0
+
+    return granted_count - terminal_count
+
+
+def is_entitlement_exercisable(
+    entitlement_id: str,
+    class_id: str,
+    reference_time_utc: Optional[datetime] = None,
```

Ruff (0.16.0)

[warning] Unused function argument: `reference_time_utc`

(ARG001)